### PR TITLE
Removes Node 10 from the matrix

### PR DESCRIPTION
**Pull request by arcanis**
**What's the problem this PR addresses?**

Node 10 will reach its end of life in april; it's probably fine to remove it now, otherwise we'll be blocked with it for at least another year, and we currently use `got` which has fairly strict "last Node" requirements (so it's likely we would be blocked from further upgrades).

**How did you fix it?**

Removed 10.x from the matrix.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by arcanis** [comment]
@nicolo-ribaudo Btw, what will Babel do regarding Node 10?

File: packages/yarnpkg-cli/sources/main.ts [hunk 1]
```diff
@@ -57,13 +57,11 @@ export async function main({binaryVersion, pluginConfiguration}: {binaryVersion:
...
   async function exec(cli: Cli<CommandContext>): Promise<void> {
     // Non-exhaustive known requirements:
-    // - 10.16+ for Brotli support on `plugin-compat`
-    // - 10.17+ to silence `got` warning on `dns.promises`
     // - 14.0 and 14.1 empty http responses - https://github.com/sindresorhus/got/issues/1496
     // - 14.10.0 broken streams - https://github.com/nodejs/node/pull/34035 (fix: https://github.com/nodejs/node/commit/0f94c6b4e4)
 
     const version = process.versions.node;
-    const range = `>=10.17 <14 || 14.2 - 14.9 || >14.10.0`;
+    const range = `>=12 <14 || 14.2 - 14.9 || >14.10.0`;
```
Comment (merceyz) [inline packages/yarnpkg-cli/sources/main.ts:64]: Should we make the minimum the LTS version?

**Comment by nicolo-ribaudo** [comment]
Oh bye bye Node 10 :wave: 

> @nicolo-ribaudo Btw, what will Babel do regarding Node 10?

We'll drop it (< 12.17 at least, which is the first one with ESM) in Babel 8. We haven't decided exactly when to release it but it will be in the next few months.

Btw, this PR doesn't impact our CI since we only run Yarn with Node.js 14.

File: packages/yarnpkg-cli/sources/main.ts (hunk 1 above)
Comment (arcanis) [reply packages/yarnpkg-cli/sources/main.ts:64]: Forgot to update - perhaps we can leave it at 12 until we get first reports that it really doesn't work on some versions.

**Pull request merged**